lib/network_utils.ml

```diff
@@ -456,8 +456,13 @@ module Linux_bonding = struct
 
 	let get_bond_master_of slave =
 		try
-			let path = Unix.readlink (Sysfs.getpath slave "master") in
-			Some (List.hd (List.rev (String.split '/' path)))
+			let master_symlink = Sysfs.getpath slave "master" in
+			let master_path = Unix.readlink master_symlink in
+			let slaves_path = Filename.concat master_symlink "bonding/slaves" in
+			let slaves = Sysfs.read_one_line slaves_path |> String.split ' ' in
```

Comment:
Could it be enough to check for the existence of `slaves_path` to conclude that this is indeed a bond slave? I think that if this path exists, we can rely on `master_path` to be what we want.